dcp-0006/dcp-0006.mediawiki

```diff
@@ -0,0 +1,395 @@
+# The TSpend is signed and broadcast to the network.
+# If the TSpend is nominally valid (Expiry in the future, valid signature, valid public key etc) it is allowed in the mempool.
+# Once the TSpend hits the mempool wallets will be notified.
+# Voting wallets with a voting policy will now start voting either yes/no or abstain at the TSpend vote window start.
+# Votes are accumulated every block. It is possible for a vote to be "short-circuited" if the outcome is already determined on a TVI block before the end of the voting interval. If an insurmountable tally of yes votes is reached, the TSpend is added to the TVI block. If an insurmountable tally of no votes is reached, the TSpend is removed from the mempool on the TVI block.
+# At the end of the TSpend window (which is always on a TVI because start is zero based) votes are tallied and if quorum and yes votes breach their respective thresholds the TSpend transaction is added to the block.
+# If the TSpend transaction does not make it into a block on the final TVI in the window the Expiry will force it from the mempool on the next block.
+
+A TSpend transaction voting window and validity is completely calculated from
+the Expiry field. The Expiry shall always be any TVI+2. This odd number is
+emergent from the handling of Expiry in the mempool. A transaction is expired
+from the Mempool one block prior to the actual Expiry. Thus in order for the
+Tspend transaction to not get evicted from the mempool on the last possible TVI
+the Expiry must be set to +2.
+
+The start of a TSpend transaction voting window is calculated as follows: Expiry -
+(TVI*TVIM) - 2.
+The end of a TSpend transaction voting window is calculated as follows: Expiry
+- 2.
+A TSpend transaction is considered "inside the window" when the block height >=
+start and block height <= end. The end parameter is inclusive for allowing a
+TSpend in a block however it is exclusive for allowing votes on the TSpend that
+may end up being allowed on the block.
+
+A treasury spend transaction is identified by the following characteristics:
+
+* The wire message transaction version shall be 3.
+* There shall be one input script.
+* There shall be at least two output scripts.
+* All output scripts shall be version 0.
+* TxIn[0] signature script shall be exactly 100 bytes in length. The script format is <code>OP_DATA_64</code> (signature), <code>OP_DATA_33</code> (Pi public key) followed by an <code>OP_TSPEND</code>.
+** The Pi public key that is included in TxIn[0] shall be of the strict compressed public key encoding type.
+* TxOut[0] script shall be an <code>OP_RETURN</code> <code>OP_DATA_32</code>.
+* TxOut[1:N] scripts shall be an <code>OP_TGEN</code> followed by a P2PKH or a P2SH script.
+
+{|
+!Transaction!!Script!!Description
+|-
+|TxIn[0]||[0]OP_DATA_64 signature<br>[65]OP_DATA_33 Pi public key<br>[99]OP_TSPEND||The value shall be sum(expenditures) + mining fee.
```

Comment:
Sorry for possibly lame question, but does this value reside outside of the script? It is not part of the allowed 100 bytes.

Reply:
Values always reside outside of the script.